pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -354,11 +355,20 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	// Remove dead container log symlinks.
 	// TODO(random-liu): Remove this after cluster logging supports CRI container log path.
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
+	now := time.Now()
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
-			err := osInterface.Remove(logSymlink)
-			if err != nil {
-				klog.Errorf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
+			if firstSeen, ok := cgc.symlinkFirstSeen[logSymlink]; ok {
```

Comment:
Is there a case when the logging symlink is restored, and the map variable would need to be deleted?

Reply:
Fixed with a comment in code.